type-definitions/immutable.js.flow

```diff
@@ -561,9 +561,78 @@ declare module 'immutable' {
     flatten(shallow?: boolean): /*this*/Map<any,any>;
   }
 
-  // OrderedMaps have nothing that Maps do not have. We do not need to override constructor & other statics
-  declare class OrderedMap<K,V> extends Map<K,V> {
+  declare class OrderedMap<K,V> extends KeyedCollection<K,V> {
+    static <K, V>(obj?: {[key: K]: V}): OrderedMap<K, V>;
+    static <K, V>(iterable: ESIterable<[K,V]>): OrderedMap<K, V>;
     static isOrderedMap(maybeOrderedMap: any): bool;
+
+    set<K_, V_>(key: K_, value: V_): OrderedMap<K|K_, V|V_>;
+    delete(key: K): this;
+    remove(key: K): this;
+    clear(): this;
+
+    update<K_,V_>(updater: (value: this) => OrderedMap<K_,V_>): OrderedMap<K_,V_>;
+    update<V_>(key: K, updater: (value: V) => V_): OrderedMap<K,V|V_>;
+    update<V_>(key: K, notSetValue: V_, updater: (value: V) => V_): OrderedMap<K,V|V_>;
+
+    merge<K_,V_>(
+      ...iterables: (ESIterable<[K_,V_]> | { [key: K_]: V_ })[]
+    ): OrderedMap<K|K_,V|V_>;
+
+    mergeWith<K_,W,X>(
+      merger: (previous: V, next: W, key: number) => X,
```

Comment:
I think that makes sense to me -- this was copy pasted from `Map` so this issue exists there as well. The other thing to note is that `iterables` should probably be `ESIterable<K_,W>` rather than `ESIterable<W>[]`.